assets/src/blocks/amp-story/index.js

```diff
@@ -62,10 +72,27 @@ export const settings = {
 	edit: EditPage,
 
 	save( { attributes } ) {
-		const { anchor, backgroundColor, mediaUrl, mediaType, poster } = attributes;
+		const {
+			anchor,
+			backgroundColor,
+			mediaUrl,
+			mediaType,
+			poster,
+			autoAdvanceAfter,
+			autoAdvanceAfterDuration,
+			autoAdvanceAfterMedia,
+		} = attributes;
+
+		let advanceAfter;
+
+		if ( [ 'auto', 'time' ].includes( autoAdvanceAfter ) && autoAdvanceAfterDuration ) {
```

Comment:
Wondering if `auto` should consider Media as well?

Reply:
I hesitated a bit to add this because I wasn't sure if it really made sense or not. It would be a bit limited as only non-looped video/audio blocks with autoplay could be taken into account.

At that point, I wonder whether the `media` option is even needed, as `auto` would be superior.

Reply:
Alternatively, we could leave the `auto` for animations only and add a description which would specify what it does (move to next slide once the animations have finished)?

> At that point, I wonder whether the media option is even needed, as auto would be superior.

It could happen in theory that the animations take longer than Media and then `media` could still have a different result being separate (if `auto` doesn't include Media).

Reply:
New issue: #1999